background/message-cache.js

```diff
@@ -30,22 +30,25 @@ export async function updateBadge(defaultStoreId) {
   const isLoggedIn = scratchAddons.globalState.auth.isLoggedIn;
   let db;
   try {
-    if (
-      scratchAddons.localState.addonsEnabled["msg-count-badge"] &&
-      (badgeSettings.showOffline || isLoggedIn) &&
-      !scratchAddons.muted
-    ) {
+    if (badgeSettings.showOffline || isLoggedIn) {
       db = await MessageCache.openDatabase();
       const count = await db.get("count", defaultStoreId);
-      // Do not show 0, unless that 0 means logged out
-      if (count || !isLoggedIn) {
-        const color = isLoggedIn ? badgeSettings.color : "#dd2222";
-        const text = isLoggedIn ? String(count) : "?";
-        // The badge will show incorrect message count in other auth contexts.
-        // Blocked on Chrome implementing store ID-based tab query
-        await promisify(chrome.action.setBadgeBackgroundColor.bind(chrome.action))({ color });
-        await promisify(chrome.action.setBadgeText.bind(chrome.action))({ text });
-        return;
+      const tooltipCount = isLoggedIn ? (count > 0 ? ` (${String(count)})` : "") : " (?)";
+      chrome.action.setTitle({
+        title: `${chrome.i18n.getMessage("extensionName")}${tooltipCount}`,
+      });
+      if (scratchAddons.localState.addonsEnabled["msg-count-badge"] && !scratchAddons.muted) {
```

Comment:
Could you remind me why this condition was moved here? The addonsEnabled check was previously in line 33.

Reply:
Now I understand. So @iqnite's original intent was to display the number on the tooltip even if `msg-count-badge` was disabled. But I believe our future plan is to only fetch the message count if necessary.

PR #7686 disabled `msg-count-badge` by default, but if we check for unread messages even if the addon is disabled, we're defeating the purpose of that PR, which was to make less requests to Scratch.